server/lib/dev_tool/fake_hydra/controllers/oauth2_controller.ex

```diff
@@ -1,13 +1,38 @@
 defmodule DevTool.FakeHydra.OAuth2Controller do
+  alias DevTool.UserServices
   use DevTool, :controller
 
   alias DevTool.FakeHydra.OAuth2Helper
+  alias DevTool.{Repo, User}
   require Logger
 
-  def auth(conn, params) do
-    uri = params["redirect_uri"]
-    code = OAuth2Helper.generate_code(params["scope"])
-    encoded_params = URI.encode_query(%{"code" => code, "state" => params["state"]})
+  def auth(conn, %{"redirect_uri" => redirect_uri, "scope" => scope, "state" => state}) do
+    users =
+      Enum.map(Repo.all(User), fn user -> user.manual_id end)
+      |> Enum.sort()
+
+    if Enum.empty?(users) do
+      {:ok, user} = UserServices.upsert_fake_user(1)
+      ^users = [1]
```

Comment:
Why are you using the pin operator "^" ? It is used to check if the variable matches with the right part but in this case the "users" variable is empty as per the if statement line 14.

Reply:
It's to override the value of the `users` var in the parent block (linter recommandation). I probably not understood it well...

Reply:
![image](https://github.com/lenra-io/devtool/assets/8969556/e1e55dc2-2478-48f5-a26e-909c1318279d)